pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +120,42 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldCap, newCap, err := util.As[*storagev1.CSIStorageCapacity](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIStorageCapacity", klog.KObj(newCap),
+	)
+
+	if oldCap == nil {
+		logger.V(4).Info("CSIStorageCapacity was created")
+		return framework.Queue, nil
+	}
+
+	if newCap.Capacity != oldCap.Capacity {
+		logger.V(4).Info("CSIStorageCapacity referenced by the Pod was created or updated, and changed Capacity", "Capacity", newCap.Capacity)
+		return framework.Queue, nil
+	}
+
+	if newCap.MaximumVolumeSize != oldCap.MaximumVolumeSize {
+		logger.V(4).Info("CSIStorageCapacity referenced by the Pod was created or updated, and changed MaximumVolumeSize", "MaximumVolumeSize", newCap.MaximumVolumeSize)
+		return framework.Queue, nil
```

Comment:
Return Queue only when either of them is _increased_?

Reply:
In addition to the case where it is increased, it seemed necessary to return Queue when it is changed to nil, so I made the correction accordingly.

https://github.com/kubernetes/kubernetes/blob/1904a56e3071035de19016cef621866a37140d5f/pkg/scheduler/framework/plugins/volumebinding/binder.go#L1050-L1057